src/xmppjs/XJSInstance.ts

```diff
@@ -430,13 +438,59 @@ export class XmppJsInstance extends EventEmitter implements IBifrostInstance {
         });
     }
 
+    private async getMucAvatar(room: string): Promise<Element> {
+        const id = uuid();
+        // check if MUC supports avatars
+        const result = await this.sendIq(new StzaIqDiscoInfo(this.myAddress.toString(), room, uuid(), "get"));
+        const supportVCards = result.getChild("query")?.getChildByAttr("var", "vcard-temp");
+        if (!supportVCards) {
+            throw Error("MUC doesn't support avatars");
+        }
+        const res = new Promise((resolve: (e: Element) => void, reject) => {
+            const timeout = setTimeout(() => reject(Error("Timeout")), 5000);
+            this.once(`iq.${id}`, (stanza: Element) => {
+                clearTimeout(timeout);
+                const vCard = (stanza.getChild("vCard") as unknown as Element);
+                if (vCard) {
+                    resolve(vCard);
+                }
+                reject(Error("Room has no avatar"));
+            });
+        });
+        log.info(`Fetching MUC Avatar of ${room}`);
+        await this.xmppSend(
+            new StzaIqVcardRequest(this.xmppAddress.toString(), room, id),
+        );
+        Metrics.remoteCall("xmpp.iq.vc2");
+        return res;
+    }
+
     public async getVCard(who: string, sender?: string): Promise<Element> {
         const id = uuid();
         const whoJid = jid(who);
         who = `${whoJid.local}@${whoJid.domain}`;
-        log.info(`Fetching vCard for ${who}`);
+        let muc = this.checkMUCCache.get(who);
```

Comment:
```suggestion
        let mucExists = this.checkMUCCache.get(who);
```
Should be a little easier to read the rest of the method :)

Reply:
Agreed.